pkg/nsx/services/ipaddressallocation/builder.go

```diff
@@ -39,7 +39,9 @@ func (service *IPAddressAllocationService) BuildIPAddressAllocation(obj metav1.O
 			return nil, fmt.Errorf("failed to find VPCInfo for IPAddressAllocation CR %s in Namespace %s", o.Name, o.Namespace)
 		}
 		ipAddressBlockVisibility = convertIpAddressBlockVisibility(o.Spec.IPAddressBlockVisibility)
-		if restoreMode && len(o.Status.AllocationIPs) > 0 {
+		if len(o.Spec.AllocationIPs) > 0 {
+			allocationIps = String(o.Spec.AllocationIPs)
```

Comment:
have you tried specifying all allocation_ips,  allocation_size and ip_address_block_visibility, can it work?

Reply:
I've tested the following cases which all work:
- 3 ipAddressBlockVisibility with allocationSize
- 3 ipAddressBlockVisibility with allocationIPs as cidr
- 3 ipAddressBlockVisibility with allocationIPs as ip